### ISSUE TITLE
Fail Harder?

### ISSUE THREAD

**Issue by mattleblanc** [UCATLAS/xAODAnaHelpers#1315]
Hi all,

I just spent a set of grid jobs making some empty trees. It's not the end of the world, but I've seen others make this mistake in the past as well: if any of the detail strings in the `TreeAlgo` has a typo in it, jobs will proceed and appear to work, but produce only empty trees at the end of the day.

It seems to me that it would be better for users and from a resource-management point of view if this algorithm crashed and burned immediately, rather than silently proceeding ...

🍻  MLB

**Comment by kratsg** [comment]
> I just spent a set of grid jobs making some empty trees. It's not the end of the world, but I've seen others make this mistake in the past as well: if any of the detail strings in the `TreeAlgo` has a typo in it, jobs will proceed and appear to work, but produce only empty trees at the end of the day.
> 
> It seems to me that it would be better for users and from a resource-management point of view if this algorithm crashed and burned immediately, rather than silently proceeding ...

Good idea. We can do some quick (and nice) parsing of the C++ source files for special algorithms.

**Comment by kkrizka** [comment]
A better solution would be for the info switch classes to throw an error if it detects a string it cannot parse.

**Comment by kratsg** [comment]
> A better solution would be for the info switch classes to throw an error if it detects a string it cannot parse.

We can't do this during config time (or at least not before the job is submitted), can we?
